### PR TITLE
Bugfix opensimAD

### PR DESCRIPTION
**Pull request by Lars-DHondt-KUL**
## Description

- use steplength bounds settings to determine whether CasADi Function is needed
- remove references to old OpenSimAD integration from testing function
- more robust handling of test function optional inputs
- improve readability of test function outcome

## Motivation and Context

- both caused errors

## How Has This Been Tested?

./Tests/test_PredSim.m

## Suggested tests for reviewers

./Tests/test_PredSim.m

### CODE REVIEW

File: Tests/runTestSimulation.m [hunk 1]
```diff
@@ -44,14 +42,11 @@
     if strcmpi(varargin{i},'run_as_batch_job')
         run_as_batch_job = varargin{i+1};
     end
-    if strcmpi(varargin{i},'PathCpp2Dll_Exe')
-        PathCpp2Dll_Exe = varargin{i+1};
-    end
     if strcmpi(varargin{i},'CasADi_path')
         CasADi_path = varargin{i+1};
     end
     if strcmpi(varargin{i},'Cpp2Dll_compiler')
```
Comment (dgupta7) [inline Tests/runTestSimulation.m:48]: should line 48 be if strcmpi(varargin{i},'OpenSimAD_compiler')?

File: Tests/runTestSimulation.m [hunk 2]
```diff
@@ -110,23 +105,22 @@
...
+if ~isempty(OpenSimAD_compiler)
+    S.Cpp2Dll.compiler = OpenSimAD_compiler;
 else
     S.Cpp2Dll = rmfield(S.Cpp2Dll,'compiler');
 end
 if max_iter > 0
     S.solver.max_iter = max_iter;
 end
 
+S.OpenSimADOptions.verbose_mode = 0; % 0 for no outputs from cmake
```
Comment (dgupta7) [inline Tests/runTestSimulation.m:122]: Might be good to have verbose mode on for tests. Or make it function input?

File: Tests/runTestSimulation.m (hunk 1 above)
Comment (Lars-DHondt-KUL) [reply Tests/runTestSimulation.m]: good catch

File: Tests/runTestSimulation.m (hunk 2 above)
Comment (Lars-DHondt-KUL) [reply Tests/runTestSimulation.m]: added it as input

**Review by dgupta7 — APPROVED** [review]
I ran the test_PredSim.m file. There were no errors.

**Pull request merged**